src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets

```diff
@@ -282,15 +283,19 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="_UpdateBlazorWasmStaticWebAssetsForV2" DependsOnTargets="GetCurrentProjectStaticWebAssets">
     <ItemGroup>
       <_CurrentWasmProjectStaticWebAssets Include="@(_ThisProjectStaticWebAssets)" Condition="'%(SourceId)' == '$(PackageId)'" />
-      <_CurrentWasmProjectStaticWebAssets>
-        <!-- We set the asset kind explicitly to build because blazor before 6.0 did its own publishing and was
-             not fully integrated with static web assets, so we let it be even when called from 6.0 targets
-        -->
-        <AssetKind>Build</AssetKind>
-      </_CurrentWasmProjectStaticWebAssets>
+    </ItemGroup>
+
+    <UpgradeLegacyWasmStaticWebAssets
+      LegacyAssets="@(_CurrentWasmProjectStaticWebAssets)"
+    >
+      <Output TaskParameter="UpgradedAssets" ItemName="_UpgradedWasmProjectStaticWebAssets" />
+    </UpgradeLegacyWasmStaticWebAssets>
+
+    <ItemGroup>
       <_ThisProjectStaticWebAssets Remove="@(_CurrentWasmProjectStaticWebAssets)" />
-      <_ThisProjectStaticWebAssets Include="@(_CurrentWasmProjectStaticWebAssets)" />
+      <_ThisProjectStaticWebAssets Include="@(_UpgradedWasmProjectStaticWebAssets)" />
```

Comment:
Legacy assets weren't update correctly (compression) and was creating issues as our logic was not detecting the compressed assets from NET5.0. 

I've moved the logic into a task to fix it and so that its simpler for us in the future to update it if needed